### DISCUSSION TITLE
cy.intercept multiple matchers found. Is it possible to only wait for the last network matcher?

### DISCUSSION THREAD

**Discussion by thaiching** [cypress-io/cypress#17299] [General]
I have a network request to spy on which will always return multiple matchers. The first few matchers usually don't contain the data that I need to perform assertion and I would need to spy only the last matcher found.

Currently I have to do a multiple cy.wait() before getting to the last matcher, which is not ideal. Just wondering if there's a better way to only wait on the last matcher?

All the interceptor network requests have the same "GET" and url pattern and the only difference are the response returned.

```
cy.intercept('GET', '**/gampad/ads?*').as('gam'); //this will usually return a few matchers 

cy.wait('@gam'). //only want to spy on the last gam request
```

**Comment by flotwig** [comment]
@thaiching `cy.wait()` waits for the requests in the order they are received, as you've noticed. However, using `.all`, this should work:

```ts
// use .all get a list of all requests then select the last one
cy.get('@gam.all').then(matches => Cypress._.last(matches))
.should(...)
```
